apps/publish/content/kill.py

```diff
@@ -184,3 +189,28 @@ def apply_kill_override(self, item, updates):
             updates.update(kill_header)
         except Exception:
             logger.exception('Failed to apply kill header template to item {}.'.format(item))
+
+    def _remove_marked_user(self, item):
+        """Remove the marked_for_user from all the published items having same 'item_id' as item being killed."""
+        item_id = item.get('_id')
+        if not item_id:
+            return
+
+        updates = {'marked_for_user': None}
+        published_service = get_resource_service(PUBLISHED)
+
+        published_items = list(published_service.get_from_mongo(req=None, lookup={'item_id': item_id}))
+        if not published_items:
+            return
+
+        for item in published_items:
```

Comment:
you could use [update_published_items](https://github.com/superdesk/superdesk-core/blob/develop/apps/publish/published_item.py#L33)

Reply:
Actually, I tried this one first, but it never returns any data, maybe those articles not present in the elastic. I tried with elastic versions 1.7 and 2.4. Not sure whats's wrong. Fetching directly from mongo does work well so I decided to do the way it is currently.